api/v1alpha1/httproute_types.go

```diff
@@ -117,13 +119,20 @@ type HTTPRouteMatch struct {
 
 	// HeaderType defines the semantics of the `Header` matcher.
 	//
+	// Support: core (Exact)
+	// Support: extended (RegularExpression)
+	// Support: custom (ImplementationSpecific)
+	//
+	// Default: "Exact"
+	//
 	// +optional
 	HeaderType *string `json:"headerType" protobuf:"bytes,3,opt,name=headerType"`
-	// Header are the Header matches as interpreted via
-	// HeaderType.
+	// Headers are the HTTP Headers to match as interpreted via
+	// HeaderType. Mutliple headers are ANDed together, meaning, a request
+	// must contain all the headers specified inorder to select this route.
 	//
 	// +optional
-	Header map[string]string `json:"header" protobuf:"bytes,4,rep,name=header"`
+	Headers map[string]string `json:"headers" protobuf:"bytes,4,rep,name=headers"`
```

Comment:
We need discussion around the type for this.
`map[string]string` seems reasonable but doesn't allow for multiple values of a header. Should this instead be `map[string][]string`?
Or should this be supported using `regex`?

Reply:
That's a good point. Are there other examples of matching one of several specified values for a header? I think `map[string][][string]` has the potential to overcomplicate other use cases. Since this could likely be accomplished with a regex and may not be a very common use case, I think I'd lean towards `map[string]string`.

Reply:
My thoughts exactly, Rob.
In my experience, users usually want to match headers against a regular expression or a specific value to make a routing decision.

We can leave this as is for now and come back to it if someone else raises this up.